contributed_definitions/nyaml/NXcoordinate_system_set.yaml

```diff
@@ -88,138 +88,259 @@ doc: |
-#            and faces no ambiguities. However, in reality typically multiple
-#            coordinate systems have to be mastered especially for complex
-#            multi-signal modality experiments.
-#          * 2 or more; If this case is realized, the best practice is that in every
-#            case where a coordinate system should be referred to the respective class
-#            has a depends_on field which resolves the possible ambiguities which specific
-#            coordinate systems is referred to. The benefit of this explicit and clear
-#            specifying of the coordinate system used in every case is that especially
-#            in combination with having coordinate systems inside deeper branches
-#            makes up for a very versatile, backwards compatible, but powerful system
-#            to express different types of coordinate systems using NeXus. In the case
-#            of two or more instances of NXcoordinate_system in one :ref:`NXcoordinate_system_set`,
-#            it is also advised to specify the relationship between the two coordinate systems by
-#            using the (NXtransformations) group within NXcoordinate_system.
-#          
-#          In effect, 1 should be the preferred choice. However, if more than one coordinate
-#          system is defined for practical purposes, explicit depends_on fields should
-#          always guide the user for each group and field which of the coordinate system
-#          one refers to.
-#     </doc>
-#     <group type="NXcoordinate_system"/>
-#     <!--How to interpret "depends_on" calls for NXtransformations instances which live
-# within an instance of NXcoordinate_system(_set)?
-# Depends on cardinality of NXcoordinate_system_set, i.e. how many NXcoordinate_system are defined?
-# - 0, the root of an NXtransformation chain i.e. "." means McStas
-# - 1, the root of an NXtransformation chain i.e. "." means "that specific CS" i.e. the one defined by the single instance of NXcoordinate_system in the set
-# - > 1, "." becomes ambiguous. My suggestion, spell out the name of the specific NXcoordinate_system instance to be used, using "."
-# despite all these, a warning should be raised, like a logic warning (and McStas kick in) or an error (i.e. be forbidden) be raised-->
-# </definition>
+  # frequently used coordinate systems with frequently relevant specializations
+  processing_reference_frame(NXcoordinate_system):
+    doc: |
+      Details about eventually relevant named directions that may
+      give reasons for anisotropies. The classical example is cold-rolling
+      where one has to specify which directions (rolling, transverse, and normal)
+      align how with the direction of the base vectors of the sample_reference_frame.
+    type(NX_CHAR):
+      doc: |
+        Type of coordinate system and reference frame according to
+        convention 1 of DOI: 10.1088/0965-0393/23/8/083501.
+      enumeration: [undefined, cartesian]
+    handedness(NX_CHAR):
```

Comment:
Many of these fields are repeating what is anyway in the base class `NXcoordinate_system`, so they can be removed even if we want to keep these specific instances of `NXcoordinate_system`.

Reply:
No these are different concepts which have no equally specific docstring counterparts currently in NXcoordinate_system_set, see my first longer point

Reply:
How is this a different concept? You are not specifying anything in the docstring what makes this handedness special.. I guess it is "the handedness of the coordinate system processing_reference_frame" vs. "the handedness of any coordinate system" (as defined in the base class). However, for using them in an appdef, there is no difference, you would put there `processing_reference_frame/handedness` anyway, so it is clear you are talking about the handedness of processing_reference_frame, even if handedness is only defined in the bases class NXcoordinate_system.

Reply:
You are right this is a point I think I forgot, processing_reference_frame inherits from NXcoordinate_system so handedness, i_alias, and i_direction fields can be removed UNLESS they specialize that concept via the docstring